modules/mcc/include/opencv2/mcc/ccm.hpp

```diff
@@ -0,0 +1,537 @@
+    CIE94_TEXTILES,
+    CIE2000,
+    CMC_1TO1,                   ///< In 1984, the Colour Measurement Committee of the Society of Dyers and Colourists defined a difference measure, also based on the L*C*h color model.
+    CMC_2TO1,
+    RGB,
+    RGBL
+};
+
+/** @brief Core class of ccm model
+
+Produce a ColorCorrectionModel instance for inference
+*/
+class CV_EXPORTS_W ColorCorrectionModel
+{
+public:
+    /** @brief Color Correction Model
+
+    Supported list of color cards:
+    - @ref Macbeth (Macbeth ColorChecker)
+    - @ref Vinyl (DKK ColorChecker)
+    - @ref DigitalSG (DigitalSG ColorChecker with 140 squares)
+
+    @param src detected colors of ColorChecker patches;\n
+                the color type is RGB not BGR, and the color values are in [0, 1];
+    @param constcolor the Built-in color card
+    */
+    CV_WRAP ColorCorrectionModel(const Mat& src, CONST_COLOR constcolor);
+
+    /** @brief Color Correction Model
+    @param src detected colors of ColorChecker patches;\n
+               the color type is RGB not BGR, and the color values are in [0, 1];
+    @param colors the reference color values,the color values are in [0, 1].\n
+    @param ref_cs the corresponding color space
+                NOTICE: For the list of color spaces supported, see the notes above;\n
+                If the color type is some RGB, the format is RGB not BGR;\n
+    */
+    CV_WRAP ColorCorrectionModel(const Mat& src, Mat colors, COLOR_SPACE ref_cs);
+
+    /** @brief Color Correction Model
+    @param src detected colors of ColorChecker patches;\n
+                the color type is RGB not BGR, and the color values are in [0, 1];
+    @param colors the reference color values,the color values are in [0, 1].
+    @param ref_cs the corresponding color space
+                NOTICE: For the list of color spaces supported, see the notes above;\n
+                If the color type is some RGB, the format is RGB not BGR;
+    @param colored mask of colored color
+    */
+    CV_WRAP ColorCorrectionModel(const Mat& src, Mat colors, COLOR_SPACE ref_cs, Mat colored);
+
+    /** @brief set ColorSpace
+
+    @note It should be some RGB color space;
+          For the list of RGB color spaces supported, see the notes above
+
+    @param cs the absolute color space that detected colors convert to;\n
+              default: sRGB
+    */
+    CV_WRAP void setColorSpace(COLOR_SPACE cs);
+
+    /** @brief set ccm_type
+    @param ccm_type the shape of color correction matrix(CCM);\n
+                    default: @ref CCM_3x3
+    */
+    CV_WRAP void setCCM_TYPE(CCM_TYPE ccm_type);
+
+    /** @brief set Distance
+
+    Supported list:
+    - @ref CIE2000
+    - @ref CIE94_GRAPHIC_ARTS
+    - @ref CIE94_TEXTILES
+    - @ref CIE76
+    - @ref CMC_1TO1
+    - @ref CMC_2TO1
+    - @ref RGB (Euclidean distance of rgb color space)
+    - @ref RGBL (Euclidean distance of rgbl color space)
```

Comment:
Not sure that it is a good idea to duplicate enum values.
Just reference on enum type is enough.

Additionally it make sense to notice about some exclusions.

Reply:
Thanks, I think you are right. I have removed enum values only if it is necessary.

Reply:
Thanks, I think you are right, and I removed enum values from the supported list only if it is necessary.